sdk/core/switcher/entry.S

```diff
@@ -274,21 +274,19 @@ __Z26compartment_switcher_entryz:
 	 * and to be run with interrupts deferred, we'd like the switcher, and
 	 * especially its stack-zeroing, to be preemtable.
 	 */
-	cincoffset        ct2, csp, -SPILL_SLOT_SIZE
 .Lswitch_entry_first_spill:
 	/*
 	 * FROM: above
 	 * ITO: .Lswitch_just_return (via .Lhandle_error_in_switcher)
 	 */
-	csc               cs0, SPILL_SLOT_cs0(ct2)
-	csc               cs1, SPILL_SLOT_cs1(ct2)
-	csc               cgp, SPILL_SLOT_cgp(ct2)
-	csc               cra, SPILL_SLOT_pcc(ct2)
-	cmove             csp, ct2
+	csc               cs0, (SPILL_SLOT_cs0-SPILL_SLOT_SIZE)(csp)
+	csc               cs1, (SPILL_SLOT_cs1-SPILL_SLOT_SIZE)(csp)
+	csc               cgp, (SPILL_SLOT_cgp-SPILL_SLOT_SIZE)(csp)
+	csc               cra, (SPILL_SLOT_pcc-SPILL_SLOT_SIZE)(csp)
```

Comment:
Unfortunately the negative offset means we don't get to use compressed `c.scsp` instruction here. `ct2` is also not accessible to the compressed `c.csc` so the instruction saved is still a net win. Remind me why we can't do the `cincoffset` before into `sp`? So that we can recover `csp` on entry in the case where the cincoffset takes it below `csp.base`?

Reply:
Thanks much for this comment.  We can, instead:
```
	csc               cs0, (SPILL_SLOT_cs0-SPILL_SLOT_SIZE)(csp)
	cincoffset        csp, csp, -SPILL_SLOT_SIZE
	csc               cs1, SPILL_SLOT_cs1(csp)
	csc               cgp, SPILL_SLOT_cgp(csp)
	csc               cra, SPILL_SLOT_pcc(csp)
```
This uses one uncompressed insruction (the first), and then compresses the rest.

Much of the rest of the switcher could stand careful analysis to move our working set into the registers available to more compressed instructions...